File: problems/2090_k_radius_subarray_averages/solution.py
class Solution:
    def getAverages(self, nums: list[int], k: int) -> list[int]:
        n = len(nums)
        result = [-1] * n
        if 2 * k + 1 > n:
            return result
        window_sum = sum(nums[:2 * k + 1])
        result[k] = window_sum // (2 * k + 1)
        for i in range(k + 1, n - k):
            window_sum += nums[i + k] - nums[i - k - 1]
            result[i] = window_sum // (2 * k + 1)
        return result

File: problems/2090_k_radius_subarray_averages/Solution.java
import java.util.*;

class Solution {
    public int[] getAverages(int[] nums, int k) {
        int n = nums.length;
        int[] result = new int[n];
        Arrays.fill(result, -1);
        int window = 2 * k + 1;
        if (window > n) return result;
        long sum = 0;
        for (int i = 0; i < window; i++) sum += nums[i];
        result[k] = (int)(sum / window);
        for (int i = k + 1; i < n - k; i++) {
            sum += nums[i + k] - nums[i - k - 1];
            result[i] = (int)(sum / window);
        }
        return result;
    }
}
